labgrid/resource/udev.py

```diff
@@ -466,6 +466,27 @@ def __attrs_post_init__(self):
         self.match['DRIVER'] = 'hub'
         super().__attrs_post_init__()
 
+@target_factory.reg_resource
+@attr.s(eq=False)
+class YKUSHPowerPort(USBResource):
+    """This resource describes a YEPKIT YKUSH switchable USB hub.
+
+    Args:
+        serial (str): serial of the YKUSH device
+        index (int): port index
+    """
+    serial = attr.ib(default="dummy", validator=attr.validators.instance_of(str))
+    index = attr.ib(default=-1, validator=attr.validators.instance_of(int),
+                    converter=int)
+
+    def filter_match(self, device):
+        match = (device.properties.get('ID_VENDOR_ID'), device.properties.get('ID_MODEL_ID'))
+
+        if match not in [("04d8", "f2f7")]:
+             return False
+
+        return super().filter_match(device)
```

Comment:
Without additional matche filters, this would match all ykush devices, even ones with different serials. So one would need to specify both serial and path in the resource configuration, which seems redundant.

Reply:
ok, now it matches against the serial only which is discoverable.